release/README.md

```diff
@@ -12,11 +12,36 @@ Istio release is currently composed of artifacts for the following repos:
 
 The release consists in retagging the artifacts and creating new annotated tags.
 
-Note that only Contributors can create a release (tags).
+Only organization members part of the [Release Engineers](https://github.com/orgs/istio/teams/release-engineers/members) team may create a release.
 
 If you are making a release from a branch, use the branch name, e.g. `BRANCH=release-0.1` for 0.1 or `master` for master.
 
-## Creating tags
+## Semi-automated Release Since 0.2
+
+Release from 0.2 onward is semi automated. It is still driven from a release engineer desktop but all actions are automated using [githubctl](https://github.com/istio/test-infra/blob/master/toolbox/githubctl/main.go), a tool of our own that acts as a github client making REST calls through the github API. The release process goes like the following:
+
+```
+# Tag the release
+$ githubctl --token_file=<github token file> \
+    --op=tagIstioDepsForRelease \
+    --base_branch=<release branch or master>
+
+# The previous command triggers rebuild and retagging on pilot, mixer and auth.
+# Wait for them to finish, then create an update PR in istio/istio
+$ githubctl --token_file=<github token file> \
```

Comment:
Yes it does. At the end of console output, it will say something like
```
2017/08/22 00:03:18 Created new PR at https://github.com/istio/istio/pull/588
```